### ISSUE TITLE
Problems with saving and loading stock TF models (Resnet50, MobileNetV3, etc.) starting from TF 2.5 

### ISSUE THREAD

**Issue by compfy-dot-com** [tensorflow/tensorflow#52978]
System: Ubuntu 20, Python 3.9.5, TF 2.5 (or 2.6)

This code: 

base_model = MobileNetV3Large(); 
base_model.save('test')

Generates a warning: "generic_utils.py:494: CustomMaskWarning: Custom mask layers require a config and must override get_config. When loading, the custom mask layer must be passed to the custom_objects argument." 

I ignored this warning, trained a model and observed that when saving and loading the model back, it is losing accuracy at which it was saved.

Expected behavior: The stock models should save and load without warnings or problems.

**Comment by tilakrayal** [comment]
@compfy-dot-com ,
In order to expedite the trouble-shooting process, could you please provide the complete code and dataset to reproduce the issue reported here.

**Comment by compfy-dot-com** [comment]
The complete code to reproduce the warning:

from tensorflow.keras.applications import MobileNetV3Large
base_model = MobileNetV3Large()
base_model.save('test')

IMHO we should focus on that warning and the reason for it: There should be none for the stock models. TF 2.4 does not seem to have this issue, it's only starting with TF 2.5.

**Comment by tilakrayal** [comment]
@compfy-dot-com ,
Can you please look at this [comment](https://github.com/tensorflow/tensorflow/issues/50647#issuecomment-877178205) from the issue with the similar warning.It helps.Thanks!

**Comment by compfy-dot-com** [comment]
The commit in question has the following comment: "Print a warning when trying to serialize an invalid custom mask." One way to interpret it is that in TF 2.5 were introduced changes that made it not backwards compatible, hence the warning. In other words, in TF 2.4 the stock models were valid and then in TF 2.5 they became invalid. If this is true, then either the stock models or the TF code need fixing; if it is not true, then the warning needs to be removed.

**Comment by tilakrayal** [comment]
@compfy-dot-com ,
The warning was added with TF 2.5 in commit [076b5be](https://github.com/tensorflow/tensorflow/commit/076b5be77d3933deb726de1fc9d954d284985b1e#diff-08518b3b0b22c5c9dc137c911fa716fe820878cb9bbf95806ffca0e0296ec7fa)

**Comment by tilakrayal** [comment]
Please post this issue on [keras-team/keras](https://github.com/keras-team/keras/issues) repo.
To know more refer to:
https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999

**Comment by compfy-dot-com** [comment]
> @compfy-dot-com , The warning was added with TF 2.5 in commit [076b5be](https://github.com/tensorflow/tensorflow/commit/076b5be77d3933deb726de1fc9d954d284985b1e#diff-08518b3b0b22c5c9dc137c911fa716fe820878cb9bbf95806ffca0e0296ec7fa)

Yes, hence see https://github.com/tensorflow/tensorflow/issues/52978#issuecomment-971681651

**Comment by compfy-dot-com** [comment]
> Please post this issue on [keras-team/keras](https://github.com/keras-team/keras/issues) repo. To know more refer to: https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999

Does this response mean that the TF team does not maintain stock models nor the compatibility with them when distributing them and it is the Keras team that should fix the models to work with the latest TF releases?

**Comment by compfy-dot-com** [comment]
What's about Google models such as MobileNetV3 - also Keras team should fix it?

**Comment by jvishnuvardhan** [comment]
@compfy-dot-com Agree with you that it was an issue with old versions.

This was resolved in recent `tf-nightly`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/6b5ef3b267d868f311f4c2ed4c003bd0/untitled1129.ipynb) is a gist for reference. If you want stable version, then it will be available in the upcoming TF2.8 in near future. Thanks!

Please verify once and close the issue if the issue got resolved for you. Thanks!
